src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -800,15 +838,193 @@ def __try_add_error(error_list, detail):
+        size_of_max_packages_allowed_in_status = self.__internal_file_capacity - status_file_without_package_list_size
+
+        while status_file_size_in_bytes > self.__internal_file_capacity:
+            # Start truncation process
+            packages_retained_in_assessment, packages_removed_from_assessment, packages_retained_in_installation, packages_removed_from_installation = \
+                self.__apply_truncation_process(self.__assessment_packages_copy, self.__installation_packages_copy, size_of_max_packages_allowed_in_status, low_pri_index)
+
+            if len(packages_removed_from_assessment) > 0:
+                # Recompose truncated status file payload (assessment)
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_package_list=packages_retained_in_assessment,
+                    truncated_substatus_msg=self.__assessment_substatus_msg_copy, substatus_index=assessment_substatus_index)
+
+            if len(packages_removed_from_installation) > 0:
+                # Recompose truncated status file payload (installation)
+                truncated_status_file = self.__recompose_truncated_status_file(truncated_status_file=truncated_status_file, truncated_package_list=packages_retained_in_installation,
+                    truncated_substatus_msg=self.__installation_substatus_msg_copy, substatus_index=installation_substatus_index)
+
+            status_file_size_in_bytes = self.__calc_status_size_on_disk(json.dumps(truncated_status_file))
+            status_file_agent_size_diff = status_file_size_in_bytes - self.__internal_file_capacity
+            size_of_max_packages_allowed_in_status -= status_file_agent_size_diff   # Reduce the max packages byte size by tombstone, new error, and escape chars byte size
+
+        self.composite_logger.log_debug("End package list truncation")
+
+        return truncated_status_file
+
+    def __split_assessment_list(self, assessment_packages):
+        """ Split package list, keep 5 minimum packages, and remaining packages for truncation """
+        min_packages_count = Constants.StatusTruncationConfig.MIN_ASSESSMENT_PACKAGE_TO_RETAIN
+        min_assessment_patches_to_retain, assessment_patches_eligible_for_truncation = (assessment_packages[:min_packages_count], assessment_packages[min_packages_count:]) \
+            if len(assessment_packages) > min_packages_count else (assessment_packages, [])
+
+        return min_assessment_patches_to_retain, assessment_patches_eligible_for_truncation
+
+    def __apply_truncation_process(self, assessment_packages, installation_packages, max_package_list_capacity, low_pri_index=None):
+        """ Truncation function call split assessment method and apply truncation on assessment and installation packages """
```

Comment:
What does it do with the split assessment? Comment does not answer that.

Suggestion for the comment: """ Function truncates packages from assessment and installation substatus's while always retaining a minimum required count of assessment packages """

Reply:
@feng-j678 please leave a note on what the status of this is.